DataFormats/Common/interface/MultiCollection.h

```diff
@@ -0,0 +1,63 @@
+// Author: Felice Pantaleo (CERN), 2025, felice.pantaleo@cern.ch
+#ifndef DataFormats_Common_MultiCollection_h
+#define DataFormats_Common_MultiCollection_h
+
+#include <ranges>
+#include <span>
+#include <utility>
+#include <vector>
+
+#include "DataFormats/Common/interface/RefProd.h"
+#include "MultiSpan.h"
+
+namespace edm {
+
+  /**
+ * @brief Lightweight persistent holder for several `edm::RefProd<Collection>`
+ *        objects.
+ *
+ * Only the vector of `RefProd`s is stored on disk.  No transient caches or
+ * synchronisation primitives live inside the class, so it is trivially
+ * movable.  Consumers obtain a flat view with `makeFlatView()`, which assembles
+ * and *returns* a fully‑populated `MultiSpan` by value.
+ *
+ * This design avoids copy/move issues with `std::once_flag` and makes the type
+ * EDM‑wrapper‑friendly while still giving fast, indexed access to the
+ * concatenated elements.
+ */
+  template <typename Collection>
+  class MultiCollection {
+  public:
+    using value_type = typename Collection::value_type;
+
+    MultiCollection() = default;
+
+    explicit MultiCollection(std::initializer_list<edm::RefProd<Collection>> refs) : refProds_{refs} {}
+
+    // ---------------- producer‑side API ----------------------------------
+    void add(edm::RefProd<Collection> const& ref) { refProds_.push_back(ref); }
+
+    // ---------------- consumer‑side helpers ------------------------------
+    /**
+   * @brief Build and return a flat view that spans all referenced collections.
+   *
+   * The returned `MultiSpan` is independent of `this`, so callers may
+   * move or store it locally without keeping the manager alive.
+   */
+    [[nodiscard]] MultiSpan<value_type> makeFlatView() const {
+      MultiSpan<value_type> ms;
+      // Framework‑managed retrieval
+      std::ranges::for_each(refProds_,
+                            [&](auto const& rp) { ms.add(std::span<const value_type>(rp->data(), rp->size())); });
```

Comment:
How about just relying on `Collection` being used as a range and use the range constructor overload of `span`?
```suggestion
                            [&](auto const& rp) { ms.add(*rp); });
```
or
```suggestion
                            [&](auto const& rp) { ms.add(std:span<const value_type>(*rp)); });
```